File: tienda-BackEnd/src/main/java/com/mi_tec/tiendaBackEnd/Security/Dto/NuevoUsuario.java
package com.mi_tec.tiendaBackEnd.Security.Dto;

import java.util.HashSet;
import java.util.Set;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

public class NuevoUsuario {

    @NotBlank
    private String nombre;
    @NotBlank
    private String nombreUsuario;
    @Email
    private String email;
    @NotBlank
    private String password;
    @NotBlank
    private String telefono;
    @NotBlank
    private String ubicacion;
    private Set<String> roles = new HashSet<>();

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getNombreUsuario() {
        return nombreUsuario;
    }

    public void setNombreUsuario(String nombreUsuario) {
        this.nombreUsuario = nombreUsuario;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = nombre;
    }
     public String getUbicacion() {
        return ubicacion;
    }

    public void setUbicacion(String ubi) {
        this.ubicacion = ubi;
    }
}

File: tienda-FrontEnd/src/app/model/nuevo-usuario.ts
export class NuevoUsuario {
  nombre!: string;
  nombreUsuario!: string;
  email!: string;
  password!: string;
  telefono!: string;
  ubicacion!: string;
  constructor(
    nombre: string,
    nombreUs: string,
    email: string,
    password: string,
    telefono: string,
    ubicacion: string
  ) {
    this.nombre = nombre;
    this.nombreUsuario = nombreUs;
    this.email = email;
    this.password = password;
    this.telefono = telefono;
    this.ubicacion = ubicacion;
  }
}
